x-pack/plugins/reporting/server/lib/esqueue/worker.js

```diff
@@ -353,6 +353,7 @@ export class Worker extends events.EventEmitter {
         constant_score: {
           filter: {
             bool: {
+              minimum_should_match: 1,
```

Comment:
We want a query that's equivalent of 
```
jobtype == this.jobtype &&
  (status == "pending" ||
    (status == "processing" && process_expiration <= nowTime))
```

I'd recommend trying to do this without using `constant_score`. Can we try:
```
{
  query: {
    bool: {
      filter: {
        bool: {
          must: { term: { jobtype: this.jobtype } },
          should: [
            { term: { status: 'pending' } },
            {
              bool: {
                must: [
                  { term: { status: 'processing' } },
                  { range: { process_expiration: { lte: nowTime } } }
                ]
              }
            }
          ]
        }
      }
    }
  }
}
```

Reply:
Dope! I still needed the `minimum_should_match: 1` in order for this to work, just an FYI

Reply:
With further digging, I find that adding `minimum_should_match: 1` was perhaps the only change needed.